server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/HeaderExtractor.java

```diff
@@ -80,10 +83,12 @@ static HeaderExtractor addressExtractor(ThrowingFunction<Mail, Address[]> addres
         };
     }
 
-    static Stream<String> toContent(Address[] addresses) {
-        return Optional.ofNullable(addresses)
-            .map(AddressHelper::asStringStream)
-            .orElse(Stream.empty());
+    static Stream<String> toAddressContents(String[] headers) {
+        return Stream.concat(
+            StreamUtils.ofNullable(headers),
+            StreamUtils.ofNullable(headers)
+                .map(Throwing.function(string -> InternetAddress.parseHeader(string, !STRICT_PARSING)))
+                .flatMap(AddressHelper::asStringStream));
```

Comment:
sorry, I don't understand this concat

Reply:
Keeping original header value, usefull in case of failure.

Long things short: Javax alter underlying value, that can not be recovered hence handling. I need to keep original value anyway as it is lost downstream.

Reply:
(If I do not do this some previously written tests fails)

Reply:
but then you are missing list of addresses with addresses, no?

Reply:
But we are in a recovery scenario.

I already added many tests and proved this PR increase feature resilience...